tester/src/test/scala/spinal/tester/scalatest/LatchTester.scala

```diff
@@ -0,0 +1,39 @@
+package spinal.tester.scalatest
+
+import org.scalatest.funsuite.AnyFunSuite
+import spinal.core._
+import spinal.core.sim._
+
+class LatchTester extends AnyFunSuite {
+  test("bad latch") {
+    SimConfig.compile(new Component {
+      val myCond   = Bool
+      val inVal    = Bits(8 bits)
+      val latchVal = Bits(8 bits)
+
+      when(myCond) {
+        latchVal := inVal
+      }
+    })
+  }
+
+  test("allowed latch") {
+    SimConfig.compile(new Component {
```

Comment:
Maybe we can add a simulation test to check the actual behavior of generated code? @Dolu1990 do you think it is needed?

Reply:
Yes it would be good. Also to see how verilator will hate us <3

Reply:
Verilator hates it. There's a warning about adding linter comments:
```
%Warning-LATCH: Users/[...]/SpinalHDL/tmp/job_1/unamed.v:14:3: Latch inferred for signal 'unamed.latchVal' (not all control paths of combinational always assign a value)
                                                                : ... Suggest use of always_latch for intentional latches
   14 |   always @(*) begin
      |   ^~~~~~
                ... Use "/* verilator lint_off LATCH */" and lint_on around source to disable this message.
%Error: Exiting due to 1 warning(s)
```
How should this be resolved? (I'm not familiar with Verilator)